pkg/app/piped/planpreview/builder.go

```diff
@@ -95,7 +95,18 @@ func newBuilder(
 	}
 }
 
-func (b *builder) Build(ctx context.Context, id string, cmd model.Command_BuildPlanPreview) ([]*model.ApplicationPlanPreviewResult, error) {
+func (b *builder) Build(ctx context.Context, id string, cmd model.Command_BuildPlanPreview) (results []*model.ApplicationPlanPreviewResult, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("an unexpected panic occurred (%w)", r)
```

Comment:
nit: If `(%w)` is more idiomatically, fine to be as is.
```suggestion
			err = fmt.Errorf("an unexpected panic occurred: %w", r)
```

Reply:
Let me keep it as it is because we are applying the same way at other places too.

Reply:
in that case that's fine